Kconfig.zephyr

```diff
@@ -34,6 +38,9 @@ source "subsys/Kconfig"
 
 source "ext/Kconfig"
 
+source "$ENV_VAR_SYM_CONTAINER_DIR/hal/*/Kconfig.module"
```

Comment:
does this work with absolute paths? I guess so, just checking

Reply:
"Why not just glob for ENV_VAR_SYM_CONTAINER_DIR/*/Kconfig.module or so? (I'm not sure how the glob syntax works in our Kconfig dialect, but I basically mean find $ENV_VAR_SYM_CONTAINER_DIR -name Kconfig.module)."

You can't do that.

If a user has cloned Zephyr to $HOME that would cause CMake to run a handful of system calls on every directory recursively down from HOME.

Having one or two directories like hal/modules creates a directory namespace that prevents this from happening (on most systems).

```
stat("/home/sebo/.config/google-chrome", {st_mode=S_IFDIR|0700, st_size=4096, ...}) = 0
lstat("/home/sebo/.config/google-chrome", {st_mode=S_IFDIR|0700, st_size=4096, ...}) = 0
open("/home/sebo/.config/google-chrome", O_RDONLY|O_NONBLOCK|O_DIRECTORY|O_CLOEXEC) = 3
fstat(3, {st_mode=S_IFDIR|0700, st_size=4096, ...}) = 0
getdents(3, /* 36 entries */, 32768)    = 1528
getdents(3, /* 0 entries */, 32768)     = 0
close(3)                                = 0

```

So, I would suggest it could look like this instead:

container_dir/zephyr
container_dir/modules/ext_dep_1
container_dir/modules/ext_dep_2
container_dir/modules/ext_dep_3

alternatively that we introduce a backwards-incompatible change that requires users to
move their git repo into a container directory. Then it could look like this:

container_dir/zephyr
container_dir/ext_dep_1
container_dir/ext_dep_2
container_dir/ext_dep_3

Reply:
Another approach;

instead of having the build system search for modules one could have CMake either ask the multi-tool what directories to use, e.g.

execute_process(tool --list-external-repo-directories EXT_REPO_DIRS)

or have the tool spit out a file that CMake is able to read. E.g.

include(${ZEPHYR_BASE}/../external_repos.cmake)

or

execute_process(PYTHON parse_repos.py ${ZEPHYR_BASE}/../manifest.xml EXT_REPOS)

Then we can have the originally desired directory layout.

Reply:
@SebastianBoe Inspired by your suggestion, I'd propose something simpler (pseudo-code), where we look for a sentinel file as was mentioned earlier. 
```
extfile=${ZEPHYR_BASE}/../zephyr_external/sentinel.cmake
if $extfile exists: include($extfile)
```
Then we would not need to run any external tool. And we would obtain complete flexibility of placement of modules/hals/etc.
For KConfig, I'm not sure. If nothing else, I suppose the sentinel.cmake could generate appropriate KConfig source paths for the modules it already knows about.

I think it would be wise to qualify the sentinel dir with "zephyr_external" or similar, rather than "hal", "modules", "external_repos.cmake" since those might actually collide.

Finally, note that not everybody uses Google's repo tool to administer multiple gits, thus the manifest.xml file can't be assumed to exist. Note that out of tree external modules might not even be gits. 
I'd like to not confuddle the build with more python scripts and assumptions on particular git-manangement tool.